Xlib horizontal wheel

ollipal opened the pull request:
Adds support for horizontal scrolling for Linux, tested on Pop!_OS 22.04.

Note that in `send_native`, y-direction must be done first, or only one direction will trigger.

I hope this helps!

Narsil commented:
Thanks for your contribution.

Don't have such a mouse at my disposal right now, but this is unlikely to break anything else so I will go ahead an merge if it passes the tests.

ollipal commented:
Ah now I figured out how to run tests locally. I'll update here if I'm able to fix it.

ollipal commented:
I also realized that there is an issue, when I tested on a view that is both horizontally and vertically scrollable at the same time (my earlier testing was done with one or another). Sorry for the broken PR! I'll attempt to fix this later today.

Narsil commented:
No worries. Take your time.

ollipal commented:
Fixed! Now in addition to implementing horizontal scrolling, I also implemented the scrolling magnitude for Linux as well.

At least on my machine, now everything works well, including the tests.
